lib/open_api_spex/open_api/map.ex

```diff
@@ -0,0 +1,395 @@
+defmodule OpenApiSpex.OpenApi.Map do
+  @moduledoc """
+  This module exposes functionality to convert an arbitrary map into a OpenApi struct.
+  """
+  alias OpenApiSpex.{
+    Components,
+    Contact,
+    Discriminator,
+    Encoding,
+    Example,
+    ExternalDocumentation,
+    Header,
+    Info,
+    License,
+    Link,
+    MediaType,
+    OAuthFlow,
+    OAuthFlows,
+    OpenApi,
+    Operation,
+    Parameter,
+    PathItem,
+    Reference,
+    RequestBody,
+    Response,
+    Schema,
+    SecurityScheme,
+    Server,
+    ServerVariable,
+    Tag,
+    Xml
+  }
+
+  def decode(%{openapi: _openapi, info: _info, paths: _paths} = map) do
```

Comment:
To be honest, since Elixir v1.6 the error messages for FunctionClauseError are quite good, as we show both the arguments and the clauses that we tried to match, so I would say a separate clause is not necessary but YMMV. :)

Reply:
@mbuhot @josevalim, I don't feel strongly either way — @mbuhot I'd defer it to you how to move forward.